### ISSUE TITLE
Option to set default date on open

### ISSUE THREAD

**Issue by TheSisb** [Hacker0x01/react-datepicker#320]
Currently in all examples the picker opens to the current date.  I want to open it to another date/year without setting a value.
Any undocumented option for this?

**Comment by rafeememon** [comment]
Hey @TheSisb, there's no hidden option for this but it sounds like a good request.  This could be an `initialDate` prop that, if specified, will set the calendar's view to this date on first load.

**Comment by uniphil** [comment]
We do this by hooking into `onFocus`, eg.:
```js
const SomeComponent = React.createClass({
  getInitialState() {
    return { selected: null };
  },
  handleDateFocus() {
    if (this.state.selected === null) {
      this.replaceState({ selected: doAnyKindOfMagicDefaultLookupHere() });
    }
  },
  handleChange(newDate) {
    this.replaceState({ selected: newDate });
  },
  render() {
    return (
      <DatePicker
          onFocus={this.handleFocus}
          onChange={this.handleChange}
          selected={this.state.selected} />
    );
  }
});
```

Which works well for us since we do some computation at focus time to determine which date should be default. However, for simpler cases this is probably annoying to have to set up.

**Comment by PetarIvancevic** [comment]
You can do this by creating a new date and pass it to moment. If you store it in your component state then you can do this:

```
const defaultDate = moment(new Date('YYYY', 'MM', 'DD'));
this.setState({ startDate: defaultDate });

...

<DatePicker selected={(this.state) ? this.state.startDate : null} />
```

Hope it helps someone.

**Comment by martijnrusschen** [comment]
@PetarIvancevic's answer looks probably like the way to go here. Do we want to document this somewhere?

**Comment by rafeememon** [comment]
@PetarIvancevic's solution sets the selected date, but @TheSisb was looking for a way without doing that.  @uniphil's code gets the right functionality but in an "unsupported" way.

**Comment by lbntly** [comment]
Hi @TheSisb, just opened a PR for this if you want to take a look https://github.com/Hacker0x01/react-datepicker/pull/486

**Comment by rafeememon** [comment]
Fixed in #486
